### PR TITLE
[FLINK-XXXX] Clear tmp directory when setting up the working directory

### PR DESCRIPTION
**Pull request by tillrohrmann**
Clear tmp directory when setting up the working directory

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointUtils.java [hunk 1]
```diff
@@ -189,18 +191,34 @@ private static void configureProcessWorkingDirectory(
                             workingDir.getAbsolutePath()));
         }
 
+        deleteTmpWorkingDirectory(workingDir);
```
Comment (slinkydeveloper) [inline flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointUtils.java]: Why the deletion of this directory fails "hard"? Can we just ignore the failure?

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointUtils.java]: Maybe. We might risk that the Flink process picks up some temporary files that it shouldn't see. That's why we are failing here. In the usual case this should hopefully not be a problem.

Comment (slinkydeveloper) [reply flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointUtils.java]: Perhaps if you want to hard fail shouldn't you check first if the folder doesn't exist? So you don't hard fail when the temp dir just doesn't exist

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointUtils.java]: I think this method won't fail if the tmp directory does not exist.

**Review by slinkydeveloper — APPROVED** [review]

**Pull request closed without merging**